src/sentry_plugins/sessionstack/static/sessionstack/contexts/sessionstack.jsx

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+const ASPECT_RATIO = 16 / 9;
+
+const SessionStackContextType = React.createClass({
+  propTypes: {
+    alias: React.PropTypes.string.isRequired,
+    data: React.PropTypes.object.isRequired
+  },
+
+  getInitialState() {
+    return {
+      showIframe: false
+    };
+  },
+
+  componentDidMount() {
+    this.parentNode = ReactDOM.findDOMNode(this).parentNode;
+    window.addEventListener('resize', this.setIframeSize, false);
+    this.setIframeSize();
+  },
+
+  componentWillUnmount() {
+    window.removeEventListener('resize', this.setIframeSize, false);
+  },
+
+  setIframeSize() {
+    if (!this.showIframe) {
+      let parentWidth = $(this.parentNode).width();
+
+      this.setState({
+        width: parentWidth,
+        height: parentWidth / ASPECT_RATIO
+      });
+    }
+  },
+
+  playSession() {
+    this.setState({
+      showIframe: true
+    });
+
+    this.setIframeSize();
+  },
+
+  render() {
+    let { session_url } = this.props.data;
+
+    if (!session_url) {
+      return (
+        <h4>Session not found.</h4>
+      );
+    }
+
+    return (
+      <div className="panel-group">
+        {this.state.showIframe ?
+          <iframe src={session_url} width={this.state.width} height={this.state.height} /> :
```

Comment:
What about making the on-premises configuration available only for on-premises deployments of Sentry? As you already said, we'll sandbox the iframe so it doesn't have access to the parent window. If people show interest for using on-premises SessionStack with hosted Sentry we'll think of a solution. We're now thinking of a way to load the player over HTTPS.